internal/subshell/sscommon/rcfile.go

```diff
@@ -199,3 +211,22 @@ func SetupProjectRcFile(templateName, ext string, env map[string]string) (*os.Fi
 
 	return tmpFile, nil
 }
+
+func absPathToCurrentExec() (string, error) {
+	exec, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	return fileutils.ResolvePath(exec)
+}
+
+func inPath(pathVal, dir string) bool {
+	paths := strings.Split(pathVal, string(os.PathListSeparator))
+	for _, path := range paths {
+		if path == dir {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I think you can dump both of these and use `fileutils.PathContainsParent` instead.

Reply:
PathContainsParent checks if the provided path is at all matching or a child of the provided parent path. It doesn't deal with the PATH envvar.

Reply:
Nor does inPath.. ? You'd have to iterate over PATH and then call `PathContainsParent `.

Reply:
I'm mostly sure that that would return false positives.

Reply:
I've renamed the function and variables to make the logic more clear. Manual testing does confirm the behavior I expect, and the added integration test should be working once I fully fix the doskey usage.

Reply:
@daved this isn't addressed

Reply:
Discussed in chat; Closing.